qiskit/circuit/library/__init__.py

```diff
@@ -181,6 +181,7 @@
    :template: autosummary/class_no_inherited_members.rst
 
    Diagonal
+   DiagonalGate
```

Comment:
I would've liked to deprecate this as the `Diagonal` circuit exists, but `DiagonalGate` seems to be used in the Aer simulators.

Reply:
imo it should be the circuit that gets removed not the gate, since circuits inherently define an eager synthesis, but that's not important.

Reply:
Agreed, maybe this would best be part of an overhaul to consistently use Gate or Circuits (probably the first) 🙂

Reply:
I agree that this is quite confusing. Maybe it's worth to add a comment somewhere about which one will be deprecated?

Reply:
I think we had the plans to deprecate `Permutation` circuit once `PermutationGate` is around. Do you think it's a good idea?